lib/processor.ts

```diff
@@ -1,44 +1,33 @@
 /**
  * @author Toru Nagashima <https://github.com/mysticatea>
  */
-'use strict'
+import type { Linter } from 'eslint'
 
-/**
- * @typedef {import('eslint').Linter.LintMessage} LintMessage
- */
-/**
- * @typedef {object} GroupState
- * @property {Set<string>} GroupState.disableAllKeys
- * @property {Map<string, string[]>} GroupState.disableRuleKeys
- */
+type LintMessage = Linter.LintMessage
+
+interface GroupState {
+  disableAllKeys: Set<string>
+  disableRuleKeys: Map<string, string[]>
+}
 
-module.exports = {
-  /** @param {string} code */
-  preprocess(code) {
+export default {
+  preprocess(code: string) {
     return [code]
   },
 
-  /**
-   * @param {LintMessage[][]} messages
-   * @returns {LintMessage[]}
-   */
-  postprocess(messages) {
+  postprocess(messages: LintMessage[][]) {
```

Comment:
Please retain explicit return types from JSDoc comments:

```suggestion
  postprocess(messages: LintMessage[][]): LintMessage[] {
```